app/src/main/java/com/ubergeek42/WeechatAndroid/service/RelayService.java

```diff
@@ -313,11 +322,26 @@ else if (e instanceof UnresolvedAddressException)
 
     @WorkerThread @Override public void onMessage(RelayMessage message) {
         kitty.trace("→ onMessage(%s)", message.getID());
-        if (state.contains(STATE.STOPPED)) return;
+        String id = message.getID();
+        SendMessageWithPromiseEvent callbackEvent = promiseCallbacks.get(id);
```

Comment:
i think `remove()` also returns the previous value so you don't need both `get()` and `remove()`